### PR TITLE
Update Webconfig

### PR DESCRIPTION
**Pull request by glumb**
Updated:
- Design
- Validation that JSON is present. Or atleast something in the textarea.
- Disable button when required field is missing
- Added Wificonfig on top and save button to the bottom to improve UX

Please test before merging - I don't have a spare ESP currently...

### CODE REVIEW

File: external_libs/wifi_config_haa/content/index.html [hunk 1]
```diff
@@ -42,42 +52,57 @@
...
+<label>
+<input type="checkbox" name="reset" value="y"> Reset HomeKit ID
+</label>
+<!-- part HTML_SETTINGS_OTA -->
+<label>
+<input type="checkbox" name="ota" value="y"> OTA Update
+</label>
+<!-- part HTML_SETTINGS_AUTOOTA -->
+<label>
+<input type="checkbox" name="autoota" value="y"> Auto OTA Updates
```
Comment (RavenSystem) [inline external_libs/wifi_config_haa/content/index.html]: This is broken. Here must be  in the middle. See original code.

**Comment by RavenSystem** [comment]
I prefer to keep JSON at top, because if there are many wifi detected, JSON will be at very bottom.

JSON can be empty for the case that you only want to change wifi settings or update firmware.

If JSON can be empty, disable button is not necessary.

File: external_libs/wifi_config_haa/content/index.html (hunk 1 above)
Comment (glumb) [reply external_libs/wifi_config_haa/content/index.html]: fixed

**Comment by glumb** [comment]
Bildschirmfoto 2019-10-21 um 21 15 36

File: external_libs/wifi_config_haa/content/index.html (hunk 1 above)
Comment (RavenSystem) [reply external_libs/wifi_config_haa/content/index.html]: Mmm, no. Must be:

`<input type="checkbox" name="autoota" value="y" <!-- part HTML_SETTINGS_AUTOOTA -->> Auto OTA Updates<br>`

But don't worry. I'll merge and the I will fix it.

Comment (glumb) [reply external_libs/wifi_config_haa/content/index.html]: It should be like this on the latest commit. This comment references the original commit.

Comment (RavenSystem) [reply external_libs/wifi_config_haa/content/index.html]: Sorry. I don't know where I saw it.

**Pull request merged**